src/docs/development/ui/layout/building-adaptive-apps.md

```diff
@@ -0,0 +1,1217 @@
+       child:  ...,
+     ),
+   ),
+ );
+}
+```
+
+The [`Shortcuts`][] widget is useful because it only
+allows shortcuts to be fired when this widget tree
+or one of its children has focus and is visible.
+
+The final option is a global listener. This listener
+can be used for always-on, app-wide shortcuts or for
+panels that can accept shortcuts whenever they're visible
+(regardless of their focus state). Adding global listeners
+is easy with [`RawKeyboard`][]:
+
+<!--skip-->
+```dart
+void initState() {
+  super.initState();
+  RawKeyboard.instance.addListener(_handleKey);
+}
+
+@override
+void dispose() {
+  RawKeyboard.instance.removeListener(_handleKey);
+  super.dispose();
+}
+```
+
+To check key combinations with the global listener,
+you can use the `RawKeyboard.instance.keysPressed` map.
+For example, a method like the following can check whether any
+of the provided keys are being held down:
+
+<!--skip-->
+```dart
+static bool isKeyDown(Set<LogicalKeyboardKey> keys) {
+ return keys.intersection(RawKeyboard.instance.keysPressed).isNotEmpty;
+}
+```
+
+Putting these two things together,
+you can fire an action when `Shift+N` is pressed:
+
+<!--skip-->
+```dart
+void _handleKey(event){
+  if(event is RawKeyDownEvent){
+     bool isShiftDown = isKeyDown({
+        LogicalKeyboardKey.shiftLeft, 
+        LogicalKeyboardKey.shiftRight,
+     |);
+     if(isShiftDown && event.logicalKey == LogicalKeyboardKey.keyN){
+        _createNewItem();
+      }
+  }
+}
+```
+
+One note of caution when using the static listener,
+is that you often need to disable it when the user
+is typing in a field or when the widget it’s associated with
+is hidden from view.
+Unlike with `Shortcuts` or `RawKeyboardListener`,
+this is your responsibility to manage. This can be especially
+important when you’re binding a Delete/Backspace accelerator for
+`Delete`, but then have child `TextFields` that the user
+might be typing in.
+
+
+[`RawKeyboard`]: {{site.api}}/flutter/services/RawKeyboard-class.html
+[`RawKeyboardListener`]: {{site.api}}/flutter/widgets/RawKeyboardListener-class.html
+
+### Mouse enter, exit, and hover
+
+On desktop, it’s common to change the mouse cursor
+to indicate the functionality about the content the
+mouse is hovering over. For example, you usually see
+a hand cursor when you hover over a button,
+or an `I` cursor when you hover over text.
+
+The Material Component set has built-in support
+for your standard button and text cursors.
+To change the cursor from within your own widgets,
+use [`MouseRegion`][]:
+
+<!--skip-->
+```dart
+return MouseRegion(
+ cursor: SystemMouseCursors.grab,
+ child: MyDraggableWidget(),
+)
+```
+
+`MouseRegion` is also useful for creating custom
+rollover and hover effects:
+
+<!--skip-->
+```dart
+return MouseRegion(
+ onEnter: (_) => setState(() => _isMouseOver = true),
+ onExit: (_) => setState(() => _isMouseOver = false),
+ onHover: (e) => print((e as PointerHoverEvent).localPosition);
```

Comment:
This line would be better written as:
`onHover: (PointerHoverEvent e) => print(e.localPosition);`

Reply:
`onHover: (PointerHoverEvent e) => print(e.localPosition),`
(the last comma)